app/controllers/users_controller.rb

```diff
@@ -1,3 +1,31 @@
 class UsersController < ApplicationController
-  def new; end
+  def show
+    @user = User.find_by(id: params[:id])
+    if @user.nil?
+      redirect_to root_url
+      flash[:danger] = t "not_found"
+    end
```

Comment:
toàn bộ code trong method này thì tách ra 1 method riêng xong dùng before action để gọi nhé, với lại chú ý sắp xếp đúng thứ tự của method